storybookjs/storybook
# React Native <0.43 support

Issue: Builds were broken for React Native versions `<0.43`. (#1553)

## What I did
Swapped out `SectionList` for `ListView`. This will be deprecated in the future (I don't believe anytime soon/immediate), so ideally we may want to export a List component that interfaces with whichever underlying `ListView` / `SectionList` implementation is supported. For now, this simply unbreaks `<0.43` builds.

There are two additional regressions: `StyleSheet.absoluteFillObject` and the use of `maxWidth` in a style. For ease of use moving forward, I added a `polyfills/` directory to address these head on.

#### `StyleSheet.absoluteFillObject`

This is only supported in recent versions. It's simply a preset style for a full screen absolutely positioned element.

#### `MinMaxViewPolyfill`

The min/max width/height features of styles are not available on earlier versions of React Native. This adds a Higher Order Component that can declaratively (through props) apply these bounding rules to any child component.


## How to test

Run the `react-native-vanilla` project and verify all continues to work.

<details><summary>Proof</summary>

![screen shot 2017-07-31 at 1 35 46 pm](https://user-images.githubusercontent.com/656630/28789915-3158f0e0-75f5-11e7-9f9d-ffe36ed6f783.png)
</details>

Closes #1553
